src/Host.cpp

```diff
@@ -1382,6 +1382,11 @@ void Host::raiseEvent(const TEvent& pE)
             mLuaInterpreter.callEventHandler(functionsList.at(i), pE);
         }
     }
+
+    // After the event has been raised but before 'event' goes out of scope,
+    // we need to safely dereference the members of 'event' that point to
+    // values in the Lua registry
+    mLuaInterpreter.freeAllInLuaRegistry(pE);
```

Comment:
@vadi2 I've moved the logic to free the data in the lua registry to inside `Host::raiseEvent`. I was thinking that this would ensure that we don't miss any events that need this logic. Let me know what you think.